File: 1022 - Sum of Root To Leaf binary Numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    
    def sumRootToLeaf(self, root: TreeNode) -> int:
        s = self.traverse(root)
        return s
    
    def traverse(self, root, b = ''):
        b = b + str(root.val)
        l, r = 0, 0
        if root.left is None and root.right is None:
            return int(b, 2)
        if root.left is not None:
            l = self.traverse(root.left, b)
        if root.right is not None:
            r = self.traverse(root.right, b)
        return l + r

File: 1022 - Sum of Root To Leaf binary Numbers.java
/**
 * Definition for a binary tree node. public class TreeNode { int val; TreeNode
 * left; TreeNode right; TreeNode() {} TreeNode(int val) { this.val = val; }
 * TreeNode(int val, TreeNode left, TreeNode right) { this.val = val; this.left
 * = left; this.right = right; } }
 */
class Solution {
    public int sumRootToLeaf(TreeNode root) {
        return traverse(root, 0);
    }
    private int traverse(TreeNode root, int s) {
        s = s * 2 + root.val;
        int l = 0, r = 0;
        if (root.left == null && root.right == null) {
            return s;
        }
        if (root.left != null) {
            l = traverse(root.left, s);
        }
        if (root.right != null) {
            r = traverse(root.right, s);
        }
        return l + r;
    }
}